### DISCUSSION TITLE
How to implement pagination using skiptoken

### DISCUSSION THREAD

**Discussion by ntnr32** [OData/WebApi#2779] [General]
How to improve pagination performance using server-side pagination technique, i.e., by using custom SkipTokenHandler.

I am looking to implement server-side pagination to improve the performance and scalability of our API, as we anticipate handling a large amount of data in our requests. I understand that using a custom SkipTokenHandler can help us achieve this goal by allowing us to control the pagination logic on the server-side.

Therefore, I would greatly appreciate it if you could provide me with some information on how to set up and configure a custom SkipTokenHandler for OData-based Web API, as well as any best practices or guidelines that you may have for implementing server-side pagination.

**Comment by habbes** [comment]
Hi @ntnr32, what version of OData are you using? You can follow this guide about server-side paging: https://learn.microsoft.com/en-us/odata/webapi-8/fundamentals/server-driven-paging. If you're specifically interested in customizing the `SkipTokenHandler`, you should also check out this tutorial: https://learn.microsoft.com/en-us/odata/webapi-8/tutorials/custom-skiptokenhandler
